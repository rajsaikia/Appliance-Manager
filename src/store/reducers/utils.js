import { cloneDeep } from 'lodash';

export const filterRooms = rooms => {
	return rooms.filter(res => res.name);
};

export const updateAppliances = (appliances, source) => {
	const newUpdateAppliances = cloneDeep(appliances);
	const orderAt = newUpdateAppliances.findIndex(o => o.id === source.id);
	const app = newUpdateAppliances.filter(res => res.id === source.id);
	const keynames = Object.keys(source);
	app[0].nickname = source.nickname;
	app[0].properties.map(res =>
		keynames.includes(res.name) ? (res.value = source[res.name]) : null
	);
	newUpdateAppliances[orderAt] = app[0];
	return newUpdateAppliances;
};

export const addNewAppliances = (appliances, payload) => {
	const newUpdateAppliances = cloneDeep(appliances);
	const data = {
		...payload,
		...{
			type: 'New Appliances',
			nickname: 'NA',
			properties: [
				{
					name: 'power',
					type: 'toggle',
					value: true,
				},
			],
		},
	};
	newUpdateAppliances.push(data);
	return newUpdateAppliances;
};

export const addAppliancesToRoom = (rooms, selectedRoom, payload) => {
	const newUpdateRooms = cloneDeep(rooms);
	newUpdateRooms
		.filter(res => res.id === selectedRoom)[0]
		.applianceIds.push(payload.id);
	return newUpdateRooms;
};
export const AddAppliancesIndex = (appliancesIndex, payload) => {
	const newUpdateRooms = cloneDeep(appliancesIndex);
	newUpdateRooms.push(payload.id);
	return newUpdateRooms;
};
