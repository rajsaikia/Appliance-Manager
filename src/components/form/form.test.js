import React from 'react';
import { render } from 'jest/utils';
import '@testing-library/jest-dom/extend-expect';
import Form from './index';

describe('<Form /> - Component', () => {
	let container = null;
	const data = {
		id: 1,
		type: 'Air Conditioner',
		nickname: 'AC',
		properties: [
			{
				name: 'power',
				type: 'toggle',
				value: true,
			},
			{
				name: 'mode',
				type: 'select',
				value: 'cool',
				options: ['cool', 'heat'],
			},
			{
				name: 'temperature',
				type: 'temperature',
				value: 60,
			},
		],
	};
	const Component = () => {
		return <Form field={data} cancelUpdate={() => {}} />;
	};
	beforeEach(async () => {
		container = render(<Component />);
	});

	afterEach(() => {
		container.unmount();
	});
	it('renders correctly', async () => {
		expect(container.asFragment()).toMatchSnapshot();
	});
	it('it should contain the text Nickname', async () => {
		expect(container.getByText('Nickname')).toBeInTheDocument();
	});
	it('it should contain the text power', async () => {
		expect(container.getByText('power')).toBeInTheDocument();
	});

	it('it should contain the text mode', async () => {
		expect(container.getByText('mode')).toBeInTheDocument();
	});
	it('it should contain the text cool', async () => {
		expect(container.getByText('cool')).toBeInTheDocument();
	});
	it('it should contain the text heat', async () => {
		expect(container.getByText('heat')).toBeInTheDocument();
	});
	it('it should contain the text temperature', async () => {
		expect(container.getByText('temperature')).toBeInTheDocument();
	});
	it('it should contain the text 60', async () => {
		expect(container.getByText('60')).toBeInTheDocument();
	});
	it('it should contain the text cool', async () => {
		expect(container.getByText('cool')).toBeInTheDocument();
	});

	it('it should contain the text F', async () => {
		expect(container.getByText('F')).toBeInTheDocument();
	});

	it('it should contain the text Apply', async () => {
		expect(container.getByText('Apply')).toBeInTheDocument();
	});
	it('it should contain the text cancel', async () => {
		expect(container.getByText('cancel')).toBeInTheDocument();
	});
});
