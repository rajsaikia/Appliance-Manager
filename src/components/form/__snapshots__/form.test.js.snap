// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> - Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="card-body"
  >
    <form>
      <div
        class="row mb-3"
      >
        <label
          class="col-sm-6 col-form-label"
          for="inputEmail3"
        >
          Nickname
        </label>
        <div
          class="col-sm-6"
        >
          <input
            name="nickname"
            value="AC"
          />
        </div>
      </div>
      <div>
        <div
          class="row mb-3"
        >
          <label
            class="form-check-label col-sm-6 col-form-label"
            for="mySwitch"
          >
            power
          </label>
          <div
            class="col-sm-6 form-check form-switch section paddingLeft"
          >
            <input
              checked=""
              class="form-check-input"
              id="selectTc"
              name="power"
              type="checkbox"
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="row mb-3"
        >
          <label
            class="col-sm-6 col-form-label"
            for="inputEmail3"
          >
            mode
          </label>
          <div
            class="col-sm-6"
          >
            <select
              class="form-select"
              name="mode"
            >
              <option
                value="cool"
              >
                cool
              </option>
              <option
                value="heat"
              >
                heat
              </option>
            </select>
          </div>
        </div>
      </div>
      <div>
        <div
          class="row mb-3"
        >
          <label
            class="col-sm-6 col-form-label"
            for="inputEmail3"
          >
            temperature
          </label>
          <div
            class="col-sm-6"
          >
            <span
              name="temperature"
            >
              60
            </span>
            <span
              class="left-margin"
            >
              F
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-plus left-margin"
              data-icon="circle-plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM256 368C269.3 368 280 357.3 280 344V280H344C357.3 280 368 269.3 368 256C368 242.7 357.3 232 344 232H280V168C280 154.7 269.3 144 256 144C242.7 144 232 154.7 232 168V232H168C154.7 232 144 242.7 144 256C144 269.3 154.7 280 168 280H232V344C232 357.3 242.7 368 256 368z"
                fill="currentColor"
              />
            </svg>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-minus left-margin"
              data-icon="circle-minus"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 256C0 114.6 114.6 0 256 0C397.4 0 512 114.6 512 256C512 397.4 397.4 512 256 512C114.6 512 0 397.4 0 256zM168 232C154.7 232 144 242.7 144 256C144 269.3 154.7 280 168 280H344C357.3 280 368 269.3 368 256C368 242.7 357.3 232 344 232H168z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="rightAlign"
      >
        <button
          class="btn btn-primary"
          type="submit"
        >
          Apply
        </button>
        <button
          class="btn btn-outline-secondary marginLeft"
          type="submit"
        >
          cancel
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
